File: Programmers/[72413] 합승 택시 요금/woong-jae/72413.js
function solution(n, s, a, b, fares) {
    const dist = Array.from(Array(n), () => Array(n).fill(Infinity));
    fares.forEach(([u, v, w]) => {
        dist[u - 1][v - 1] = w;
        dist[v - 1][u - 1] = w;
    });
    
    for(let i = 0; i < n; i++) dist[i][i] = 0;
    
    for(let k = 0; k < n; k++) {
        for(let i = 0; i < n; i++) {
            for(let j = 0; j < n; j++) {
                dist[i][j] = Math.min(dist[i][j], dist[i][k] + dist[k][j]);
            }
        }
    }
    
    let answer = Infinity;
    for(let i = 0; i < n; i++) {
        answer = Math.min(answer, dist[s - 1][i] + dist[i][a - 1] + dist[i][b - 1]);
    }
    
    return answer;
}

File: Programmers/[72413] 합승 택시 요금/ONE/72413.java
import java.util.*;

class Solution {
    private int[][] weights;
    public int solution(int n, int s, int a, int b, int[][] fares) {
        int answer = Integer.MAX_VALUE;
        weights = new int[n + 1][n + 1];

        for (int[] w : weights)
            Arrays.fill(w, 100001 * n);

        for (int i = 1; i <= n; i++)
            weights[i][i] = 0;

        for (int[] fare : fares) {
            int u = fare[0], v = fare[1], w = fare[2];
            weights[u][v] = w;
            weights[v][u] = w;
        }

        for (int k = 1; k <= n; k++)
            for (int i = 1; i <= n; i++)
                for (int j = 1; j <= n; j++)
                    if(weights[i][j] > weights[i][k] + weights[k][j])
                        weights[i][j] = weights[i][k] + weights[k][j];

        for (int i = 1; i <= n; i++)
            answer = Math.min(weights[s][i] + weights[i][a] + weights[i][b], answer);

        return answer;
    }
}